tests/main/interfaces-many-core-provided/task.yaml

```
@@ -55,6 +54,16 @@ execute: |
     echo "For each core-provided slot"
     SNAP_MOUNT_DIR="$(os.paths snap-mount-dir)"
     for plugcmd in "$SNAP_MOUNT_DIR"/bin/"$CONSUMER_SNAP".* ; do
+
+        # Just connect 30% of the interfaces on debian 10
+        # Debian 10 has bad performance disconnecting interfaces
+        # and the test fails (kill-timeout) trying either to remove
+        # interfaces or removing the snap
+        if os.query is-debian-10 && [ "${RANDOM: -1}" -ge 3 ]; then
```

Manpage states that RANDOM is in range 0-32768, so the test skips 99% of the interfaces? Perhaps you wanted

```suggestion
        # 10922 is 1/3 of the range produced by RANDOM
        if os.query is-debian-10 && [ "${RANDOM: -1}" -ge 10922 ]; then
```

---

@bboozzoo "${RANDOM: -1}" gives you the last number of the RANDOM value

---

Duh, ok, I'm totally confused now. So `${RANDOM:-1}` evaluates to a number in range 0-32768, with the default being -1 if unset. However `${RANDOM: -1}` (notice the space), evaluates to a number in range 0-9, so it's like `$((RANDOM % 10))`? I can't find any reference to that in the bash manpage. I think this warrants a comment at least. I'm not even sure that's a documented behavior though.

If you wish to keep the numbers low, I think this would be a better option `[ "$((RANDOM % 3))" = 0 ]`

---

Sure, I'll update to [ "$((RANDOM % 3))" = 0 ] it is much more clear

Thanks for the suggestion

---

Woah, shell is the best! Looks like this is in the man-page as `${parameter:offset}` (i.e. Substring Expansion. I think that `${RANDOM: -1}` is equivalent to `${RANDOM:0-1}` which makes it a bit more clear but it's still super confusing. I would use something like `echo ${RANDOM}|cut -b1` here to be way more explicit.